# HG changeset patch
# User Francois Bissey <francois.bissey@canterbury.ac.nz>
# Date 1305759178 -43200
# Node ID f4d90fd5493d56fd18c39aeed61915c21008c62c
# Parent  502d88d966cb338774726af2a7963ca393aae4e2
#11244 fix more deprecation warnings. Enable in matrix/constructor.py.

diff --git a/sage/graphs/base/graph_backends.py b/sage/graphs/base/graph_backends.py
--- a/sage/graphs/base/graph_backends.py
+++ b/sage/graphs/base/graph_backends.py
@@ -492,12 +492,7 @@
             doctest:...
             <class 'sage.graphs.base.graph_backends.NetworkXGraphDeprecated'>
         """
-        import warnings
         from sage.misc.misc import deprecation
-        warnings.warn("Your graph object is saved in an old format since networkx "+
-                      "was updated to 1.0.1. You can re-save your graph by typing "+
-                      "graph.save(filename) to make this warning go away.",
-                      DeprecationWarning, stacklevel=2)
         deprecation("Your graph object is saved in an old format since networkx "+
                     "was updated to 1.0.1. You can re-save your graph by typing "+
                     "graph.save(filename) to make this warning go away.")
@@ -565,12 +560,7 @@
             doctest:...
             <class 'sage.graphs.base.graph_backends.NetworkXDiGraphDeprecated'>
         """
-        import warnings
         from sage.misc.misc import deprecation
-        warnings.warn("Your digraph object is saved in an old format since networkx "+
-                      "was updated to 1.0.1. You can re-save your digraph by typing "+
-                      "digraph.save(filename) to make this warning go away.",
-                      DeprecationWarning, stacklevel=2)
         deprecation("Your digraph object is saved in an old format since networkx "+
                     "was updated to 1.0.1. You can re-save your digraph by typing "+
                     "digraph.save(filename) to make this warning go away.")
diff --git a/sage/matrix/constructor.py b/sage/matrix/constructor.py
--- a/sage/matrix/constructor.py
+++ b/sage/matrix/constructor.py
@@ -2516,6 +2516,7 @@
             if ncols is None:
                 if n.is_square():
                     import warnings
+                    warnings.resetwarnings()
                     warnings.warn("invocation of block_matrix with just a list whose length is a perfect square is deprecated. See the documentation for details.", DeprecationWarning, stacklevel=2)
                     nrows = ncols = n.sqrt()
                 else:
